### PR TITLE
multipass, util: introduce util.env_cmd module

### PR DESCRIPTION
**Pull request by cjp256**

### CODE REVIEW

File: craft_providers/util/env_cmd.py [hunk 1]
```diff
@@ -0,0 +1,50 @@
...
+    :param ignore_environment: Start with an empty environment.
+
+    :returns: List of env command strings.
+    """
+    env_cmd = ["env"]
+
+    if ignore_environment:
+        env_cmd.append("-i")
+
+    if env is None:
```
Comment (facundobatista) [inline craft_providers/util/env_cmd.py:41]: Does it make sense to ignore the environment, and also set env vars?

I assume it doesn't. What do you think  about `assert env is None` in the block above, and directly return env_cmd without ending the `if`? That way we cover for a potential misuse of the function by mistake.

**Review by facundobatista — CHANGES_REQUESTED** [review]
Looks good! Added small comment.

File: craft_providers/util/env_cmd.py (hunk 1 above)
Comment (cjp256) [reply craft_providers/util/env_cmd.py:41]: It does if you want env to mask the default environment and initialize env with only with what you specify.  I've personally used it in a number of cases in the past.

Whether or not we'll use it here, I am unsure.  I just added it for completeness should we want to.

**Comment by cjp256** [comment]
Rebased, and dropped the empty "env" usage if env is None.

File: craft_providers/util/env_cmd.py (hunk 1 above)
Comment (facundobatista) [reply craft_providers/util/env_cmd.py:41]: Ah, it ignores the *previous* environment. Ok.

**Review by facundobatista — APPROVED** [review]
Awesome!

**Review by sergiusens — APPROVED** [review]

**Pull request merged**